site/app/views/AutoGradingView.php

```diff
@@ -627,17 +625,12 @@ public function showPeerResults(TaGradedGradeable $ta_graded_gradeable, bool $re
         $annotation_path = FileUtils::joinPaths($this->core->getConfig()->getCoursePath(), 'annotations', $gradeable->getId(), $id, $active_version);
         $annotated_file_names = [];
         if (is_dir($annotation_path) && count(scandir($annotation_path)) > 2) {
-            $first_file = scandir($annotation_path)[2];
-            $annotation_path = FileUtils::joinPaths($annotation_path, $first_file);
-            if (is_file($annotation_path)) {
-                $dir_iter = new \DirectoryIterator(dirname($annotation_path . '/'));
-                foreach ($dir_iter as $fileinfo) {
-                    if (!$fileinfo->isDot()) {
-                        $no_extension = preg_replace('/\\.[^.\\s]{3,4}$/', '', $fileinfo->getFilename());
-                        $pdf_info = explode('_', $no_extension);
-                        $pdf_id = implode("_", array_slice($pdf_info, 0, -1));
-                        if (file_get_contents($fileinfo->getPathname()) != "") {
-                            $pdf_id = $pdf_id . '.pdf';
+            if (is_dir($annotation_path)) {
+                $dir_iter = new \FilesystemIterator(dirname($annotation_path));
+                foreach ($dir_iter as $dir_info) {
+                    if ($dir_info->isDir() && count(scandir($dir_info->getPathname()))) {
+                        if (file_get_contents($dir_info->getPathname()) !== "") {
```

Comment:
This makes no sense to me. Why are you checking for empty directories and then attempting to do a `file_get_contents` on said empty directory (which would error with `Is a directory`)?

Reply:
Rewritten to be similar to how PDFController manages annotations, removing the unneeded checks above